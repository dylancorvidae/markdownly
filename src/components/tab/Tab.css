.Tab {


  input {
    position: absolute;
    left: -99999em;
    top: -99999em;
    z-index: 1;
  
    & + label {
      cursor: pointer;
      float: left;
      border: 1px solid #aaa;
      border-bottom: 0;
      background-color: #fff;
      margin-right: -1px;
      padding: .5em 1em;
      position: relative;
          
      &:hover {
        background-color: #eee;
      }
    }
  
    &:checked + label {
      box-shadow: 0 3px 0 -1px #fff,
      inset 0 5px 0 -1px #428bca;
      background-color: rgb(255, 168, 236);
      border: solid 1px #428bca;
      z-index: 2;
    }
  }
  button {
    z-index: 3;
    margin-right: 5px
  }
}
