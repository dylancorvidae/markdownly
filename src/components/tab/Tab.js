import React from 'react';
import PropTypes from 'prop-types';

const Tab = ({ title, handleDelete }) => {
  return (
    <li>
      <input type="radio" value={title} name="tabs"></input>
      <label htmlFor={title}> TAB {title}</label>
      <button value={title} onClick={() => handleDelete(title)}>X</button>
    </li>
  );
};

Tab.propTypes = {
  title: PropTypes.string.isRequired,
  handleDelete: PropTypes.func.isRequired
};

export default Tab;
