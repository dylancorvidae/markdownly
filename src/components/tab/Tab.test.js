import React from 'react';
import { shallow } from 'enzyme';
import Tab from './Tab';

describe('Tab component', () => {
  it('renders Tab', () => {
    const wrapper = shallow(<Tab title={'1'} handleDelete={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
