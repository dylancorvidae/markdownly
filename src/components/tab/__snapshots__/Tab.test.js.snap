// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component renders Tab 1`] = `
<li
  className="Tab"
>
  <input
    id="1"
    name="tabs"
    type="radio"
    value="1"
  />
  <label
    htmlFor="1"
  >
     TAB 
    1
  </label>
  <button
    onClick={[Function]}
    value="1"
  >
    X
  </button>
</li>
`;
